resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/Client.scala

```diff
@@ -216,8 +216,70 @@ private[spark] class Client(
                 .endContainer()
               .endSpec()
             .done()
-          submitCompletedFuture.get(30, TimeUnit.SECONDS)
-        }
+          try {
+            submitCompletedFuture.get(30, TimeUnit.SECONDS)
```

Comment:
One of the error cases we hit was that the pod was stuck in the pending state because the image could not pull. We want to count the spark-submit as complete once we are able to actually start the Spark job itself.